PCIe 6.0 Packet Generator and Exerciser/PCIe 6.0 Packet Generator and Exerciser/Endpoint/configuration_visitor_classes/configuration_write_visitor.cpp

```diff
@@ -0,0 +1,103 @@
+    if (current->getRegisterType() == 0 || current->getRegisterType() == 1)
+    {
+        configuration->setReceivedMasterAbortBit(); // Setting the Received Master Abort Bit in the Status Register as we received an Unsupported Request (UR)
+        return 0; // Cpl with UR
+    }
+
+    // In case the sent Register Number belongs to a Read Write Register, then you can modify its value
+    switch (registerNumber)
+    {
+    case 2: // Command Register
+        current->setRegisterValue((current->getRegisterValue() & (current->getRegisterMask())) | (data & ~(current->getRegisterMask()))) ;
+        break;
+
+    case 3: // Status Register
+        // You can't write to the Signaled System Error bit and the SERR# Enable bit in the Command Register is = 0
+        if((data & 0x00004000) && !(command->getRegisterValue() & 0x00000100))
+            data &= 0xFFFFBFFF;
+
+        // You can't write to the Master Data Parity Error bit and the Parity Error Response bit in the Command Register is = 0
+        if((data & 0x00000100) && !(command->getRegisterValue() & 0x00000040))
+            data &= 0xFFFFFEFF;
+
+        // Then we can write to the Status Register
+        current->setRegisterValue((current->getRegisterValue() & (current->getRegisterMask())) | (data & ~(current->getRegisterMask())));
+        break;
+
+    case 7: // BAR0
+        current->setRegisterValue((current->getRegisterValue() & (current->getRegisterMask())) | (data & ~(current->getRegisterMask())));
+        break;
+
+    case 8: // BAR1
+        current->setRegisterValue((current->getRegisterValue() & (current->getRegisterMask())) | (data & ~(current->getRegisterMask())));
+        break;
+
+    case 9: // BAR2
+        current->setRegisterValue((current->getRegisterValue() & (current->getRegisterMask())) | (data & ~(current->getRegisterMask())));
+        break;
+
+    case 10: // BAR3
+        current->setRegisterValue((current->getRegisterValue() & (current->getRegisterMask())) | (data & ~(current->getRegisterMask())));
+        break;
+
+    case 15: // Expansion ROM Base Address Register
+        if(configuration->isMemorySpaceEnabled()) // if the Memory space is enabled, so you can write in the bits (0, 11->31)
+        {
+            current->setRegisterValue((current->getRegisterValue() & (current->getRegisterMask())) | (data & ~(current->getRegisterMask())));
+        }
+        else // if the Memory space is disabled, so you can't write in this register
+        {
+            configuration->setReceivedMasterAbortBit(); // Setting the Received Master Abort Bit in the Status Register as we received an Unsupported Request (UR)
+            return 0; // Cpl with UR
+        }
+        break;
+    }
+
+    return 1; // The Write is done Successfully
+}
+
+unsigned int ConfigurationWriteVisitor::visitPcieCapabilityStructure(PCIECapability * capability)
+{
+    /* To be implemented when the PCIE Capability structure is finished */
```

Comment:
I already added a comment here before, this will not be merged yet until implementation is completed.